kong/schema_service.go

```diff
@@ -2,13 +2,18 @@ package kong
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"net/http"
 )
 
 // AbstractSchemaService handles schemas in Kong.
 type AbstractSchemaService interface {
 	// Get fetches an entity schema from Kong.
 	Get(ctx context.Context, entity string) (Schema, error)
+	// Validate validates an arbitrary entity in Kong.
+	// Used for custom entities, or entities that does not have a Validate method in the corresponding service.
+	Validate(ctx context.Context, entityType string, entity interface{}) (bool, string, error)
```

Comment:
Just food for thought: `entityType` passed as an arbitrary string is quite error-prone - users can pass e.g. `service` instead of `services`. That is not obvious what are the allowed entity types here. We could define an enum that would include all the allowed entities and map them to the endpoint name (`services`, `vaults`, etc.). On the other hand, I understand this has to allow arbitrary names as it has to support custom entities with names we do not know upfront - from this perspective, a rigid enum doesn't make sense.

We could meet somewhere between and define const strings for the built-in types, still accepting an arbitrary `entityType string`. The question is whether this is something we need, taking into account that we already have strongly typed variants of `Validate` in particular built-in entities' service interfaces.

Reply:
Defined `EntityType` and added constants for some known types.